Description: Defined _XSERVER64 in header
---
Bug-Debian: https://bugs.debian.org/750375
Forwarded: https://github.com/hime-ime/hime/pull/496
--- hime-0.9.10.orig/src/im-client/hime-im-client.c
+++ hime-0.9.10/src/im-client/hime-im-client.c
@@ -25,9 +25,6 @@
 #include <unistd.h>
 #include <signal.h>
 #include <errno.h>
-#ifndef _XSERVER64
-#define _XSERVER64
-#endif
 #include "hime.h"
 #include "hime-protocol.h"
 #include "hime-im-client.h"
--- hime-0.9.10.orig/src/im-client/hime-im-client.h
+++ hime-0.9.10/src/im-client/hime-im-client.h
@@ -17,6 +17,11 @@
 
 #ifndef HIME_IM_CLIENT_H
 #define HIME_IM_CLIENT_H
+
+#ifndef _XSERVER64
+#define _XSERVER64
+#endif
+
 struct HIME_PASSWD;
 
 typedef struct HIME_client_handle_S {
